### PR TITLE
TF: rework XLA generate tests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

In the light of recent findings (https://github.com/huggingface/transformers/issues/16838), this PR reworks existing XLA generate tests. The following key changes were made:
1. Added a `@unittest.skipIf` on XLA generate tests, to skip when no GPU is present;
2. Rework XLA `sample` tests -- due to the minor numerical differences that arise when we use XLA (and that we can't control), the sampling step will gather different samples even when we use the same seed. The only thing we can properly test is whether a) we can seed them and b) the results are sensible;
3. Adds at least one XLA test where the batch size is > 1 and the inputs have different lengths, so we can confirm that masking works (GPT-2 is not working 💔 , added a TODO);
4. Removes redundant tests (we had tests outside the integration tests that were testing the same thing).

### CODE REVIEW

File: tests/t5/test_modeling_tf_t5.py [hunk 1]
```diff
@@ -496,7 +480,10 @@ def test_greedy_xla_generate_simple(self):
         output_strings = tokenizer.batch_decode(output_ids, skip_special_tokens=True)
         output_strings_xla = tokenizer.batch_decode(output_ids_xla, skip_special_tokens=True)
 
-        expected_output_string = ["Heute ist ein schöner Tag."]
+        expected_output_string = [
+            "Heute ist ein schöner Tag.",
+            "Ich habe vier Katzen, drei Hunde, zwei Vögel und ein Pferd.",
+        ]
```
Comment (Rocketknight1) [inline tests/t5/test_modeling_tf_t5.py:486]: Are we guaranteed that the logits are stable enough that we'll always sample this exact output? A flaky test can be really annoying!

Comment (gante) [reply tests/t5/test_modeling_tf_t5.py:486]: We haven't had problems with similar tests, so I'm assuming we won't have problems :D

Comment (Rocketknight1) [reply tests/t5/test_modeling_tf_t5.py:486]: ...fair point!

**Review by Rocketknight1 — APPROVED** [review]
Overall, this looks great! The fact that we can't really test sample outputs is annoying, but I see why we can't really get around that problem.

File: tests/gpt2/test_modeling_tf_gpt2.py [hunk 2]
```diff
@@ -294,21 +294,6 @@ def create_and_check_gpt2_lm_head(self, config, input_ids, input_mask, head_mask
         result = model(inputs)
         self.parent.assertEqual(result.logits.shape, (self.batch_size, self.seq_length, self.vocab_size))
 
-    def create_and_check_gpt2_xla_generate(self, config, input_ids, *args):
```
Comment (patrickvonplaten) [inline tests/gpt2/test_modeling_tf_gpt2.py:297]: Not super happy with deleting the whole test. Could we add a skipif statement for now and correct it later?

File: tests/t5/test_modeling_tf_t5.py [hunk 3]
```diff
@@ -227,23 +227,6 @@ def create_and_check_t5_decoder_model_past_large_inputs(
         # test that outputs are equal for slice
         tf.debugging.assert_near(output_from_past_slice, output_from_no_past_slice, rtol=1e-3)
 
-    def create_and_check_t5_xla_generate(self, config, input_ids, *args):
```
Comment (patrickvonplaten) [inline tests/t5/test_modeling_tf_t5.py:230]: same -> think XLA should be run at least once on CPU for every PR. Let's add a skipif with a TODO statement?

**Review by patrickvonplaten — APPROVED** [review]

File: tests/gpt2/test_modeling_tf_gpt2.py [hunk 4]
```diff
@@ -408,10 +393,6 @@ def test_gpt2_lm_head(self):
         config_and_inputs = self.model_tester.prepare_config_and_inputs()
         self.model_tester.create_and_check_gpt2_lm_head(*config_and_inputs)
 
-    def test_gpt2_xla_generate(self):
```
Comment (patrickvonplaten) [inline tests/gpt2/test_modeling_tf_gpt2.py:411]: @gante - do we have a fast XLA test for GPT2 somewhere else?

**Comment by gante** [comment]
@patrickvonplaten reintroduced the fast tests, will merge as soon as CI gets to green
